src/core/algorithm/graphToMarkovChain.test.js

```diff
@@ -203,6 +228,33 @@ describe("core/algorithm/graphToMarkovChain", () => {
       expect(normalize(osmc)).toEqual(normalize(expected));
     });
 
+    it("works on a simple chain with a loop edge", () => {
+      // The loop edge will implicitly correspond to one forward edge with weight 2, and one backward edge with weight 1
+      const loop = edge("loop", n1, n1);
+      const e1 = edge("e1", n1, n2);
+
+      const g = new Graph().addNode(n1).addNode(n2).addEdge(e1).addEdge(loop);
+      const edgeWeight = () => ({forwards: 2, backwards: 1});
+      const osmc = createOrderedSparseMarkovChain(
+        createConnections(g, edgeWeight, 0.0)
+      );
+      const expected = {
+        nodeOrder: [n1.address, n2.address],
+        chain: [
+          {
+            neighbor: new Uint32Array([0, 1]),
+            // Reproduce floating point imprecision
```

Comment:
We generally try to exactly representable numbers for tests like this,
so that you don’t have to be quite so careful here. Here, the
un-normalized out-flow from `n1` is `2 * f + b`, and the un-normalized
out-flow from `n2` is `b`. How about taking forward and backward weights
of `f = 3` and `b = 2` so that all the un-normalized out-weights are
powers of 2 and thus integer multiples of their reciprocals are exactly
representable? Thus:

```diff
diff --git a/src/core/algorithm/graphToMarkovChain.test.js b/src/core/algorithm/graphToMarkovChain.test.js
index f4f49d03..844e475b 100644
--- a/src/core/algorithm/graphToMarkovChain.test.js
+++ b/src/core/algorithm/graphToMarkovChain.test.js
@@ -234,7 +234,7 @@ describe("core/algorithm/graphToMarkovChain", () => {
       const e1 = edge("e1", n1, n2);
 
       const g = new Graph().addNode(n1).addNode(n2).addEdge(e1).addEdge(loop);
-      const edgeWeight = () => ({forwards: 2, backwards: 1});
+      const edgeWeight = () => ({forwards: 3, backwards: 2});
       const osmc = createOrderedSparseMarkovChain(
         createConnections(g, edgeWeight, 0.0)
       );
@@ -243,12 +243,11 @@ describe("core/algorithm/graphToMarkovChain", () => {
         chain: [
           {
             neighbor: new Uint32Array([0, 1]),
-            // Reproduce floating point imprecision
-            weight: new Float64Array([2 / 5 + 1 / 5, 1]),
+            weight: new Float64Array([5 / 8, 1]),
           },
           {
             neighbor: new Uint32Array([0, 1]),
-            weight: new Float64Array([2 / 5, 0]),
+            weight: new Float64Array([3 / 8, 0]),
           },
         ],
       };
```